### ISSUE TITLE
Issues with mouse movement

### ISSUE THREAD

**Issue by Goldfire345** [Vwing/daggerfall-unity-1#1]
Hello there 
Os Android 13
Device lenovo m9
 hello there i recently installed the app and i'm having issues with mouse input the issue being that i don't think the mouse support is properly implemented in the app.The reason i believe this is because the mouse doesn't smoothly move around  allowing the character to turn fully in one motion instead that doesn't happen it just stops when the cursor hits the edge of the screen. Its a sign that the app treats the mouse movement as touch input. and not proper mouse input.  Ive had experience with several other source ports (open touch gamings delta quad, zeta and alpha touch as well as open mw) all of those let you turn around fully in one motion with the mouse, so i can say for sure its not my mouse or my tablet or aside from that the app works pretty good. if something can be done about that it would be great. i'll attach a recording of my screen showing the mouse issues.  the last bit i can add is this happens off a fresh install so no mods or any major settings changes the version of daggerfall that im using is the one from the uesp site with the default options checked on the installer

https://github.com/Vwing/daggerfall-unity-android/assets/52189902/eac806eb-84b2-43d4-8790-4978e0debd5a

**Comment by Vwing** [comment]
Thank you for bringing this to my attention! I'll take a look, and hopefully will have a fix in the next release.

**Comment by Vwing** [comment]
@Goldfire345 Want to play test this release candidate of v1.1.1.2 DFU Android? I may have fixed your mouse input issue.

64bit apk: https://drive.google.com/file/d/1n-fLyxnjaX1UdI3R7KSCh0Fi-8TdP0DI/view?usp=sharing

32bit apk: https://drive.google.com/file/d/1Qmy03l7LiIunFybf1vmSqlaZIx9oeCrF/view?usp=sharing

Changelist:

* Fixed resolution and quality sliders breaking the game (Thanks to referencing @sandstranger's own implementation of an Android port https://github.com/sandstranger/daggerfall-unity)
* Implemented framerate slider in Advanced > Video settings
* Implemented ‘RunToggle’ action, which is distinct from ‘AutoRun’.
    * Instead of forcing you to run forward, it simply keeps you in the run state. So you can move in any direction or stop moving altogether. It is now the default ‘Run’ button on on-screen controls, though you can switch it back to ‘Run’ with the button binding dropdown.
* Fixed joystick bindings conflicting with on-screen bindings, causing dead inputs
* On-screen joystick input is now consistent across all screen resolutions
* On-screen joysticks now take up exactly half the screen, with no dead input space in the middle.
* Resolution slider now has 10 steps
* Separated out joystick, mouse, and on-screen input so that all may be enabled simultaneously
* Reenabled resolution setting stage of new app installation flow
* Connected mouse input might be fixed
    * 'might' because I don’t have a bluetooth mouse to test it.
* Switched from Vulkan to OpenGLES3, which should increase the number of compatible devices, especially lower-end devices.
    * An extra bonus is that OpenGLES3 seems to perform better than Vulkan even on high-end devices.

**Comment by Vwing** [comment]
I believe this is fixed in [v1.1.1.2](https://github.com/Vwing/daggerfall-unity-android/releases/tag/v1.1.1.2). Feel free to re-open the issue, though, if it continues to be a problem @Goldfire345

**Comment by Vwing** [comment]
Nevermind, another user brought to my attention that this issue is still not fixed. Looking into it, I think this unity forum thread holds the answer https://forum.unity.com/threads/android-pc-like-mouse-gaming-input-with-requestpointercapture.1228902/

I'll most likely implement the fix later this week.

**Comment by ImCrus** [comment]
@Vwing
Thank you! And sorry to bother you... didn't properly read the post then, my bad. 
Do you know an estimate date to when you will apply the fix? I'm quite intrigued to properly try the port! Once again, thank you a lot!

**Comment by Vwing** [comment]
I'm not giving any estimates at the moment, but "soon". I've been swamped for a couple weeks with contract work, and today I finally have time to start addressing this issue and other problems users have reported to me.

**Comment by Vwing** [comment]
@ImCrus  This issue proved to be more difficult than expected. I will fix it, but it's going to have to wait for the next release.

The problem is with Unity's default behavior on Android. It doesn't call the functions it's supposed to be calling when locking the cursor. The fix is going to require making a plugin that calls those functions, as [this person had](https://forum.unity.com/threads/android-pc-like-mouse-gaming-input-with-requestpointercapture.1228902/) (and is now charging [$15 for the solution](https://assetstore.unity.com/packages/tools/input-management/mobile-input-capture-221017)).

**Comment by ImCrus** [comment]
@Vwing Just take your time, I'm sure a lot of people won't mind waiting a lil while longer for quality. You're doing a massive project and it's really surprising the fact you're doing this at all, plus the quality to the extent that you're doing it.

And I'm really sorry to hear that :/  I hope it's not too much of a kick in the head. I don't really know how it all works and I really wish I could help In anyway possible but I'm in University so that complicates things too :') 
So please, take your time, and thank you for your hard work!

**Comment by ImCrus** [comment]
Hi,
@Vwing I tried it right now and sadly it sitll happens, idk if it's copium or something, but it feels better but it still happens...
If I can somehow help, let me know!

**Comment by Vwing** [comment]
> Hi,
> @Vwing I tried it right now and sadly it sitll happens, idk if it's copium or something, but it feels better but it still happens...
> If I can somehow help, let me know!
> 

@ImCrus 

Yeah, it's still not done. If you want, you could implement the fix and submit a Pull Request. That's... just about the only way to help in this instance.

**Comment by amlenny** [comment]
Is this still intended to be fixed? I really like the idea of playing on my tablet, but would greatly prefer mouse controls to touch controls.

I still primarly play on my PC, which limits the amount I can play this. I usually have my tablet with me, but the touch controls aren't doing it for me.

**Comment by Vwing** [comment]
Yes, I intend to fix it. I recently created a plugin that should fix it, and I just need to hook it in with the DFU Android code.

Like I thought, it was a fairly complicated fix for what seems like a minor issue. The biggest hurdle is done, so hopefully will have an update with the fix implemented soon.

**Comment by amlenny** [comment]
I'm super thankful that you are still working on fixing this. I see some
recent releases that don't have this in the patch notes yet.

I did read more carefully how you documented the issue in your release
notes for early versions, and I wanted to clarify that the game is
completely unplayable with a mouse currently. It's not that you can't move
more than 180 with a single swipe. It's that you are locked to a small
field of view much less than 180° until you pause to reset your screen
position. So if you're in a fight and the enemy moves more than a little
bit you have to pause to reposition. It's super finicky and I haven't been
able to defeat the first skeleton with this limitation. You can't track
with them you have to pause and try to catch up, but once they get off
screen the first time it's very hard.

Walking around the first dungeon I was able to touch the screen to move
beyond that limited range, but switching my right hand between the mouse
and the touch screen and trying to fight is literally impossible.

Thanks again for your awesome work making this game available on our phones
and cheap Android tablets!

**Comment by Vwing** [comment]
@Goldfire345 @ImCrus @amlenny latest release fixes this issue: https://github.com/Vwing/daggerfall-unity-android/releases/tag/v1.1.1.8

If you try it out, please let me know how it goes.
